server.js

```diff
@@ -1,35 +1,92 @@
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
-mongoose.Promise = Promise;
-
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
+import listEndpoints from "express-list-endpoints";
+import startDB, { getBookModel } from "./database/database";
+import { dbErrorHandler } from "./middleware/dbErrorHandler";
+
 const port = process.env.PORT || 8080;
 const app = express();
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
+app.use(dbErrorHandler);
 
-// Start defining your routes here
+//Router setup
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const landing = {
+    about: "Welcome to My Book Store📚",
+    APIs: listEndpoints(app),
+  };
+  res.send(landing);
+});
+
+app.get("/books", async (req, res) => {
+  const { limit, title } = req.query;
+
+  const bookModel = getBookModel();
+  try {
+    let data = await bookModel.find({}).exec();
+    let success = true;
+    let error = "";
+
+    if (title) {
+      //Partial search
+      data = await bookModel
+        .find({
+          $or: [{ title: { $regex: title, $options: "i" } }],
+        })
+        .exec();
+
+      if (!data.length) {
+        success = false;
+        error = "No result found";
+        return res.status(404).json({ data, success, error });
+      }
+    }
+
+    if (limit) {
+      const numLimit = Number(limit);
+      if (numLimit < 0 || !Number.isInteger(numLimit)) {
+        success = false;
+        error = "Invalid limit value";
+      } else {
+        data = await bookModel.find().limit(limit);
+      }
+    }
+
+    return res.status(200).json({ data, success, error });
+  } catch (error) {
+    return res.status(400).json({ error: error.toString() });
+  }
 });
 
-// Start the server
-app.listen(port, () => {
-  console.log(`Server running on http://localhost:${port}`);
+app.get("/books/:bookId", async (req, res) => {
+  const bookID = Number(req.params.bookId);
+
+  let data = {};
+  let success = true;
+  let error = "";
+
+  if (bookID < 0 || !Number.isInteger(bookID)) {
+    res.status(404).json({ data, success: false, error: "Invalid bookId" });
+  }
+  try {
+    data = await getBookModel().find({ bookID }).exec();
+
+    if (!data) {
+      data = {};
+      success = false;
+      error = "No result found";
+      return res.status(404).json({ data, success, error });
```

Comment:
Really like thar u have more then one error message, good for the sure to se it in the frontend